edb/schema/name.py

```diff
@@ -39,7 +39,7 @@ def __new__(
         cls: Type[NameT],
         name: Union[SchemaName, str],
         module: Optional[str] = None,
-    ) -> NameT:
+    ) -> Any:
```

Comment:
This change doesn't seem to be correct.

Reply:
It looked wrong to me, too. I did it because, if left "NameT", mypy complains this way:

![image](https://user-images.githubusercontent.com/2576032/72289445-71ec9a80-364b-11ea-8184-fe4cd0b0828e.png)

`edb/schema/name.py:70:9: error: Returning Any from function declared to return "NameT"`

How should it be fixed? It looks like # type: ignore gives Any.

Reply:
Should I remove the type annotation from cls as well?

Reply:
No, keep the `cls` annotation. We need the `TypeVar` here to make `UnqualifiedName.__new__` happy.  Use `cast` on the result: `cast(NameT, result)`

Reply:
I think so, since I cannot use `isinstance` in this case, with a generic.